CrowdStrike/glide-core
# Use minify-literals instead of minify-html-literals

## 🚀 Description

Dependabot picked up on a problematic dependency, https://github.com/kangax/html-minifier, from **2022**. The reason we pull in `html-minifier` is because of our dependency on https://github.com/asyncLiz/minify-html-literals. We [use minify-html-literals](https://github.com/CrowdStrike/glide-core/blob/main/packages/components/esbuild.js#L31-L37) to minify in our build process. 

Having briefly looked at that [CVE](https://nvd.nist.gov/vuln/detail/CVE-2022-37620), it's a regex ddos vulnerability which doesn't seem like a huge issue (despite the high CVE score).  Since it's code we are producing ourselves and only minifying in the build process, it's likely low risk. We are checking in with security still.  But even so, it appears  `minify-html-literals` is no longer maintained:

- https://github.com/asyncLiz/minify-html-literals/issues/52
- https://github.com/asyncLiz/minify-html-literals/issues/54
- https://github.com/asyncLiz/minify-html-literals/issues/56

Same for `html-minifier` - there hasn't been a release since [2019](https://github.com/kangax/html-minifier/releases/tag/v4.0.0) and seems [abandoned](https://github.com/kangax/html-minifier/issues/1151) ( [1](https://github.com/kangax/html-minifier/issues/1137), [2](https://github.com/kangax/html-minifier/issues/1135) ).

At any rate, I stumbled on [`minify-literals`](https://www.npmjs.com/package/minify-literals).  It is a fork of `html-minifer`, but uses [`html-minifier-terser`](https://github.com/explodingcamera/esm/blob/main/packages/minify-literals/package.json#L36C4-L36C24) instead, which does not have the vulnerability listed above.  It also [resolved open issues with `html-minifier`](https://github.com/explodingcamera/esm/tree/main/packages/minify-literals#credits).

I reviewed [all of the changes](https://github.com/explodingcamera/esm/commits/main/packages/minify-literals) to the best of my ability and think it looks good.  I also spot-compared the output in a few files from using this and the other library. The only difference was variable name changes.

So even though we may be okay using `html-minifer`, it's no longer up to date and means there are additional risks there.  This package is new, and may fall into the same category eventually.  It is [MIT licensed](https://github.com/explodingcamera/esm/blob/main/packages/minify-literals/package.json#L15), so we could eventually pull it into our own repo at some point and maintain it going forward if we'd like.

Thoughts?


## 📋 Checklist



- I have read and followed the [Contributing Guidelines](https://github.com/crowdstrike/glide-core/blob/main/CONTRIBUTING.md).
- I have added tests to cover new or updated functionality.
- I have created or updated stories in Storybook to document the new functionality.
- I have included a changeset with this Pull Request if it adds/updates/removes functionality for consumers.
- I have scheduled a Design Review for these changes, if one is required.
- I have followed the [ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/) and/or met with the Accessibility Team to ensure this functionality is accessible.

## 🔬 How to Test

- Take a snapshot of your `dist/` so you can reference it later
- Pull down this branch
- Run `pnpm start:production` from `packages/components`
- Review the differences to double-check nothing crazy is going on



## 📸 Images/Videos of Functionality

N/A
